cmd/kube-scheduler/app/server.go

```diff
@@ -363,207 +106,23 @@ through the API as necessary.`,
 	return cmd
 }
 
-// SchedulerServer represents all the parameters required to start the
-// Kubernetes scheduler server.
-type SchedulerServer struct {
-	SchedulerName                  string
-	Client                         clientset.Interface
-	InformerFactory                informers.SharedInformerFactory
-	PodInformer                    coreinformers.PodInformer
-	AlgorithmSource                componentconfig.SchedulerAlgorithmSource
-	HardPodAffinitySymmetricWeight int32
-	EventClient                    v1core.EventsGetter
-	Recorder                       record.EventRecorder
-	Broadcaster                    record.EventBroadcaster
-	// LeaderElection is optional.
-	LeaderElection *leaderelection.LeaderElectionConfig
-	// HealthzServer is optional.
-	HealthzServer *http.Server
-	// MetricsServer is optional.
-	MetricsServer *http.Server
-}
+func Run(c schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}) error {
+	// To help debugging, immediately log version
+	glog.Infof("Version: %+v", version.Get())
 
-// NewSchedulerServer creates a runnable SchedulerServer from configuration.
-func NewSchedulerServer(config *componentconfig.KubeSchedulerConfiguration, master string) (*SchedulerServer, error) {
-	if config == nil {
-		return nil, errors.New("config is required")
-	}
+	// Apply algorithms based on feature gates.
+	// TODO: make configurable?
+	algorithmprovider.ApplyFeatureGates()
 
 	// Configz registration.
-	// only register if we're actually exposing it somewhere
-	if len(config.MetricsBindAddress) > 0 || len(config.HealthzBindAddress) > 0 {
-		if c, err := configz.New("componentconfig"); err == nil {
-			c.Set(config)
-		} else {
-			return nil, fmt.Errorf("unable to register configz: %s", err)
-		}
-	}
-
-	// Prepare some Kube clients.
-	client, leaderElectionClient, eventClient, err := createClients(config.ClientConnection, master)
-	if err != nil {
-		return nil, err
+	if cz, err := configz.New("componentconfig"); err == nil {
```

Comment:
why is `if len(config.MetricsBindAddress) > 0 || len(config.HealthzBindAddress) > 0`  removed for this if block?

Reply:
It does not hurt to set the configz struct if no http server is installed. This follows the pattern e.g. found in the controller manager.